Purchases/Purchasing/StoreKitAbstractions/StoreProduct.swift

```diff
@@ -186,6 +190,52 @@ public extension StoreProduct {
 
 }
 
+@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.2, *)
+public extension StoreProduct {
+    /// Asynchronously determines the subset of ``discounts`` that's eligible for the current user.
+    /// - Parameter purchases: the instance of `Purchases` to use when determining eligibility.
+    /// - Seealso: ``discounts``
+    /// - Note: if checking for eligibility for a `StoreProductDiscount` fails (for example, if network is down),
+    /// that discount will fail silently and be considered not eligible.
+    func eligibleDiscounts(withPurchases purchases: Purchases) async -> [StoreProductDiscount] {
+        let discounts = self.discounts
+        let product = self
+
+        return await withTaskGroup(of: Optional<StoreProductDiscount>.self) { group in
+            for discount in discounts {
+                group.addTask {
+                    do {
+                        let eligibility = try await purchases.checkPromotionalDiscountEligibility(
+                            forProductDiscount: discount,
+                            product: product
+                        )
+
+                        return eligibility.isEligible ? discount : nil
```

Comment:
since we end up checking if nil later anyway, maybe it'd be clearer to just return `eligibility.isEligible` and check that?